website/src/documentation/patterns/side-inputs.md

```diff
@@ -46,29 +54,36 @@ For instance, the following code sample uses a `Map` to create a `DoFn`. The `Ma
 {% github_sample /apache/beam/blob/master/examples/java/src/main/java/org/apache/beam/examples/snippets/Snippets.java tag:SideInputPatternSlowUpdateGlobalWindowSnip1
 %}
 ```
+```py
+No sample present.
+```
 
 
 ## Slowly updating side input using windowing
 
-You can read side input pcollection periodically into distinct windows.
-Later, when you apply side input to your main input, windows will be matched automatically 1:1.
-This way, you can guarantee side input consistency on the duration of the single window.
+You can read side input data periodically into distinct PCollection windows.
+Later, when you apply the side input to your main input, each main input
+window is automatically matched to a single side input window.
+This guarantees consistency on the duration of the single window,
+meaning that each window on the main input will be matched to a single
+version of side input data.
 
-To do this, you can utilize PeriodicSequence PTransform that will generate infinite sequence
-of elements with some real-time period:
+Described approach can be implemented using combination of
+PeriodicSequence or PeriodicImpulse PTransforms and SDF Read or ReadAll
+PTransforms:
 
-1. Use the PeriodicImpulse transform to generate windowed periodic sequence.
-
-    a. MAX_TIMESTAMP can be replaced with some closer boundary if you want to stop generating elements at some point.
+1. Use the PeriodicImpulse or PeriodicSequence PTransform to generate an
```

Comment:
Oh! Can we format this with bullet points:

1. Use the PeriodicImpulse... to:

* Generate an infinite sequence of elements...
* Assign the elements to separate windows...

1. Read data...